### ISSUE TITLE
Release Candidate - "SDL2_image.framework is damaged and can't be opened

### ISSUE THREAD

**Issue by kjblanchard** [libsdl-org/SDL_image#284]
Hello!  Thanks much for the work on the m1 Mac fixes for this.  I went ahead and was using the release candidate [2.5.2](https://github.com/libsdl-org/SDL_image/releases/tag/prerelease-2.5.2), and am able to add includes and link via the sdl2_image-config.cmake file, thanks a lot for including that as well.

However, when running I get a "SDL2_image.framework is damaged and can't be opened" when running the binary with m1.  The cmake output I get is "SDL2_image.framework/Versions/A/SDL2_image' (no such file), `... '/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' (code signature in <1A56DFBF-1899-326F-8290-2BCBEB28500F> '/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' not valid for use in process: library load disallowed by system policy), '/System/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' (no such file)"`

Snippets of cmake that I'm using:
```
find_package(sdl2_image CONFIG REQUIRED)
...
target_include_directories(supergoon_engine
        PUBLIC
        "${CMAKE_CURRENT_SOURCE_DIR}/"
        SDL2_image::SDL2_image
        )
...
target_link_libraries(supergoon_engine PUBLIC SDL2 fmod fmodstudio SDL2_image::SDL2_image)
```

I don't get any errors when building at all and I am including and using the library in the code.

**Comment by madebr** [comment]
`SDL2_image::SDL2_image` is a target, which you can only link to.
It cannot be used in `target_include_directories` (which only except directories).

The target `SDL2_image::SDL2_image` contains both the necessary compile flags + link flags.


The cmake package should be used as:
```cmake
find_package(SDL2_Image CONFIG REQUIRED)
...
target_link_libraries(supergoon_engine PUBLIC SDL2_image::SDL2_image)
```

**Comment by kjblanchard** [comment]
Thanks for the quick response!  My mistake on the include directory.  Even after removing it, and modifying the casing of the names to match yours, I'm still running into the same issue.  Still trying debugging it on my end.  

On another note, it seems really strange but downloading the .dmg file (as I wanted to download again just incase), there is a random 'f' char on line 44 of the sdl2_image-config.cmake file.

![image](https://user-images.githubusercontent.com/51925034/174459245-23e52a0b-c0f1-4915-b6ca-08941098aa03.png)

**Comment by madebr** [comment]
Thanks, removed the stray `f` in https://github.com/libsdl-org/SDL_image/commit/d68b9caf7d6050953044fb2dd907efac4fe1962d.
I wasn't able to test the cmake config scripts for the Macos frameworks so some breakage was expected.
(I would really appreciate it if you could test the pre-releases of all SDL libraries (SDL2/SDL2_image/SDL2_mixer/SDL2_ttf)

The error looks something Xcode specific, which I have no experience with.
```
... '/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' (code signature in <1A56DFBF-1899-326F-8290-2BCBEB28500F> '/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' not valid for use in process: library load disallowed by system policy), '/System/Library/Frameworks/SDL2_image.framework/Versions/A/SDL2_image' (no such file)
```

**Comment by kjblanchard** [comment]
Sure, thanks a lot.

I think I got this working.  I modified the  system policy for terminal, and noticed was able to get a different error `Library not loaded: @rpath/SDL2.framework/Versions/A/SDL2` I was building SDL2 from source with add_subdirectory in my cmake project instead of installing it with the dmg. I went ahead and downloaded the RC for sdl2 which has the cmake packageconfig, and then it started working. 

I did test it with ttf as well, initially I was getting the same issue but after grabbing SDL2 both are working as expected

**Comment by madebr** [comment]
Interesting, I didn't know SDL_image's framework sdk only works with a SDL2 framework sdk and not with a vendored SDL2.
So you're either all-in or all-out.

So is your issue resolved and can this issue be closed?

For posterity sake, what system policy did you need to modify?
Perhaps another Macos user can comment on whether this is the correct solution.

**Comment by kjblanchard** [comment]
Yeah, I'm not sure if it was due to the difference between vendor/framework, or was due to me not having a sdl2-config.cmake file in /Libraries/Frameworks since I was building from source.  I was using a FindSDL2.cmake file after building it from source so that find_library would work.

Sure, here is the specific setting I changed.  Although, after changing this setting it didn't "fix" the problem it just allowed me to see a better error code.  I flipped them back to off afterwards.

Thanks a lot for the help, I'll go ahead and close this.

![image](https://user-images.githubusercontent.com/51925034/174460456-90e0e6e5-13ce-410e-a9b9-9df8211937da.png)
